### PR TITLE
Refactor schema package

### PR DESCRIPTION
**Pull request by ArmanMazdaee**
**What this PR does / why we need it**:
- Refactor jsonschema generation
- Refactor example generation
- Implement basic uischema generation
- Refactor schema package

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -709,6 +650,6 @@
       "type": "array"
     }
   },
-  "title": "Resumic Schema",
+  "title": "Resumic resume Schema",
```
[the remark this answers is not in the document]
Comment (ArmanMazdaee) [reply schema.json]: I was not sure if we gonna have some other schema in the feature so I had made the title more specificity. I could change it back if you think it's more appropriate.

File: example.json [hunk 2]
```diff
@@ -185,8 +183,8 @@
     {
       "language": "English",
       "score": {
-        "type": "objective",
-        "value": "fluent"
+        "type": "GPA",
```
[the remark this answers is not in the document]
Comment (ArmanMazdaee) [reply example.json]: As right now, the examples are made from schema's tags and the score has a single struct the example generated could have just a single value. If we want to have different example values I think the only way is to have two different structs. Also, we could make score example something more general like "percent".

File: schema/schema.go [hunk 3]
```diff
@@ -1,162 +1,162 @@
 package schema
```
[the remark this answers is not in the document]
Comment (ArmanMazdaee) [reply schema/schema.go]: I agree that maybe we need to separate them in the future but I think right now there is no need. Do you want me to rename the placeholder tag to the example so this way in the future we just can add a placeholder tag and use that for placeholder data?

File: schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply schema.json]: Ok. Just wanted to know the reasoning. No other schema planned for now, but ok with keeping it as is.

File: example.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply example.json]: I think we have another issue with moving to percent or something similar like:
type, min, max and value. So we have the actual value, but also the range of the type, which gives us the possibility to calculate percentages and maybe compare resumes a bit easier internationally.

File: schema/schema.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply schema/schema.go]: Yeah let's rename this to example as it makes it clearer, what this data currently is. Happy to keep using it as the placeholder for now.

**Pull request merged**